syntax = "proto3";

package kvstore;

message PutRequest {
  string key = 1;
  string value = 2;
  int64 ttl_ms = 3; // 0 = no TTL
}
message PutResponse {
  bool success = 1;
  string error = 2;
}

message GetRequest { string key = 1; }
message GetResponse {
  bool found = 1;
  string value = 2;
  int64 ttl_ms = 3;
  string error = 4;
}

message RemoveRequest { string key = 1; }
message RemoveResponse { bool success = 1; string error = 2; }

service KVStore {
  rpc Put (PutRequest) returns (PutResponse);
  rpc Get (GetRequest) returns (GetResponse);
  rpc Remove (RemoveRequest) returns (RemoveResponse);
}
